app/policies/portfolio_item_policy.rb

```diff
@@ -43,6 +42,10 @@ def can_read_and_update_destination?(destination_id)
       rbac_access.resource_check('update', destination_id, Portfolio)
   end
 
+  def portfolio_id
+    @record.class == Portfolio ? @record.id : @record.portfolio_id
```

Comment:
@eclarizio Since this is a PortfolioItemPolicy would it ever get passed in a Portfolio object. Couldn't this just be
```
def portfolio_id
 @record.order_id
end
```

Reply:
@eclarizio create? seems to be the odd ball case here.

Reply:
@mkanoor 

> @eclarizio Since this is a PortfolioItemPolicy would it ever get passed in a Portfolio object. Couldn't this just be
> 
> ```
> def portfolio_id
>  @record.order_id
> end
> ```

You're right, the only oddball is `#create?`, but basically when it comes in as a call from `#user_capabilities`, it will be for the specific portfolio item, that is the reasoning for this logic. Otherwise, if it comes in from the controller, we pass in the portfolio specifically.